mesos-simulation/src/main/scala/mesosphere/mesos/simulation/SimulatedDriver.scala

```diff
@@ -0,0 +1,111 @@
+package mesosphere.mesos.simulation
+
+import java.util
+
+import akka.actor.{ ActorRef, ActorSystem, Props }
+import com.typesafe.config.{ Config, ConfigFactory }
+import org.apache.mesos.Protos._
+import org.apache.mesos.SchedulerDriver
+import org.slf4j.LoggerFactory
+
+import scala.collection.JavaConversions._
+
+/**
+  * The facade to the mesos simulation.
+  *
+  * It starts/stops a new actor system for the simulation when the corresponding life-cycle methods of the
+  * [[SchedulerDriver]] interface are called.
+  *
+  * The implemented commands of the driver interface are forwarded as messages to the [[DriverActor]].
+  * Unimplemented methods throw [[NotImplementedError]]s.
+  */
+class SimulatedDriver(driverProps: Props) extends SchedulerDriver {
+
+  private[this] val log = LoggerFactory.getLogger(getClass)
+
+  private[this] def driverCmd(cmd: AnyRef): Status = {
+    driverActorRefOpt match {
+      case Some(driverActor) =>
+        log.debug(s"send driver cmd $cmd")
+        driverActor ! cmd
+      case None =>
+        log.debug("no driver actor configured")
+    }
+    status
+  }
+
+  override def declineOffer(offerId: OfferID): Status =
+    driverCmd(DriverActor.DeclineOffer(offerId))
+
+  override def launchTasks(offerIds: util.Collection[OfferID], tasks: util.Collection[TaskInfo]): Status =
+    driverCmd(DriverActor.LaunchTasks(offerIds.toSeq, tasks.toSeq))
+
+  override def killTask(taskId: TaskID): Status = driverCmd(DriverActor.KillTask(taskId))
+  override def reconcileTasks(statuses: util.Collection[TaskStatus]): Status = {
+    driverCmd(DriverActor.ReconcileTask(statuses.toSeq))
+  }
+
+  override def declineOffer(offerId: OfferID, filters: Filters): Status = ???
```

Comment:
I think we should allow partial implementations. I do not want to simulate all aspects of Mesos. So what's the alternative to `???`?